src/main.sh

```diff
@@ -10,6 +10,11 @@ source src/fetcher.sh
 source src/util_functions.sh
 
 function main() {
+  if [[ "${INTERACTIVE_MODE}" == "true" ]]; then
+    echo -e "Running in interactive mode...\n"
+    check_toml_env
+  fi
+
```

Comment:
this part of code some how contradicts with  what is done in the ci. 

there, the intention is to read from env during schedule (purely non-interactive use). but here, we're introducing it as to be read in interactive_mode.

it is better to update the documentation to ask the user to export variables directly? not sure. will change later. currently, we can go ahead with this change as user do not need to export individual variables.